docs/testing/e2e-plan.md

```diff
@@ -0,0 +1,79 @@
+# Curated Packages End-To-End Testing Plan
+
+This document summarizes the current state and future plans for end-to-end (E2E) testing in the EKS-A Curated Packages (CP) project.
```

Comment:
every time I see "CP" I think control plane

Reply:
> every time I see "CP" I think control plane

I've been trying to avoid saying "packages" alone, because it feels so generic, but you make a great point about control plane too. :/ Ugh. Cpackages seems awkward. I dunno.

Reply:
I think "package" is quite fine, even if generic.